### PR TITLE
added technical json to filings

### PR DESCRIPTION
**Pull request by thorwolpert**
*Issue #:* /bcgov/entity#6618

*Description of changes:*
- added technical json feature to model & core
- fixed lint issue in reports

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by thorwolpert** [comment]
skipping SonarCloud code smell as its duplicated text in a test

File: legal-api/src/legal_api/core/filing.py [hunk 1]
```diff
@@ -246,3 +250,21 @@ def get_filings_by_status(business_id: int, status: [], after_date: date = None)
...
+
+        Returns: {
+            List: or None of the Legal Filing JSON segments.
+            }
+        """
+        if not (filing := self.json):
+            return None
+
+        legal_filings = []
+        for k in filing['filing'].keys():  # pylint: disable=unsubscriptable-object
```
Comment (jordiwes) [inline legal-api/src/legal_api/core/filing.py:265]: is k an id, and if so, can we name it so?

File: legal-api/src/legal_api/models/filing.py [hunk 2]
```diff
@@ -127,12 +127,41 @@ class Source(Enum):
     }
 
     __tablename__ = 'filings'
+    # this mapper is used so that new and old versions of the service can be run simultaneously,
+    # making rolling upgrades easier
+    __mapper_args__ = {
```
Comment (jordiwes) [inline legal-api/src/legal_api/models/filing.py:132]: Where is it used?

File: legal-api/src/legal_api/core/filing.py (hunk 1 above)
Comment (thorwolpert) [reply legal-api/src/legal_api/core/filing.py:265]: the code is moved over from the model.Filings, and k is the key from keys().

File: legal-api/src/legal_api/models/filing.py (hunk 2 above)
Comment (thorwolpert) [reply legal-api/src/legal_api/models/filing.py:132]: it's used by SQLAlchemy as part of the framework.
i'll expand on the comments here.

**Review by jordiwes — APPROVED** [review]
Great, thanks LGTM

**Pull request merged**